cli/src/pcluster/resources/imagebuilder/update_and_reboot.yaml

```diff
@@ -149,13 +256,47 @@ phases:
                 else
                   package-cleanup -y --oldkernels --count=1
                 fi
+                # Install kernel-devel during OS update, so that headers are aligned with new kernel. 
+                # The same is done for Debian through `apt-get -y install linux-aws`
                 yum -y install kernel-devel
```

Comment:
I'm not sure this is always correct.

In RHEL derivatives if you pin the kernel to the version currently available in the AMI (via `DISABLE_KERNEL_UPDATE`), but the AMI is old enough, then this command may pull a version of `kernel-devel` newer than the current kernel, wouldn't it?

I'm afraid in this case for safety we should install a specific version via `yim install kernel-devel-${<current_version>}`, where `<current_version>` is derived from `uname -r`.

Reply:
Agreed to implement in a separate PR.